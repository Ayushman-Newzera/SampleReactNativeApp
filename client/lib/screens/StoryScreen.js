var _interopRequireDefault=require("@babel/runtime/helpers/interopRequireDefault");Object.defineProperty(exports,"__esModule",{value:true});exports.default=void 0;var _slicedToArray2=_interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));var _react=_interopRequireWildcard(require("react"));var _reactNative=require("react-native");var _ProgressBar=_interopRequireDefault(require("../utilities/ProgressBar"));var _jsxFileName="/Users/ayushman/Desktop/SampleReactNativeApp1/client/src/screens/StoryScreen.js";function _getRequireWildcardCache(nodeInterop){if(typeof WeakMap!=="function")return null;var cacheBabelInterop=new WeakMap();var cacheNodeInterop=new WeakMap();return(_getRequireWildcardCache=function _getRequireWildcardCache(nodeInterop){return nodeInterop?cacheNodeInterop:cacheBabelInterop;})(nodeInterop);}function _interopRequireWildcard(obj,nodeInterop){if(!nodeInterop&&obj&&obj.__esModule){return obj;}if(obj===null||typeof obj!=="object"&&typeof obj!=="function"){return{default:obj};}var cache=_getRequireWildcardCache(nodeInterop);if(cache&&cache.has(obj)){return cache.get(obj);}var newObj={};var hasPropertyDescriptor=Object.defineProperty&&Object.getOwnPropertyDescriptor;for(var key in obj){if(key!=="default"&&Object.prototype.hasOwnProperty.call(obj,key)){var desc=hasPropertyDescriptor?Object.getOwnPropertyDescriptor(obj,key):null;if(desc&&(desc.get||desc.set)){Object.defineProperty(newObj,key,desc);}else{newObj[key]=obj[key];}}}newObj.default=obj;if(cache){cache.set(obj,newObj);}return newObj;}function StoryScreen(_ref){var route=_ref.route,navigation=_ref.navigation;var _route$params=route.params,stories=_route$params.stories,handleBorderActivity=_route$params.handleBorderActivity;var userStories=stories.stories;var _useState=(0,_react.useState)(0),_useState2=(0,_slicedToArray2.default)(_useState,2),curStory=_useState2[0],setCurStory=_useState2[1];var useProgress=function useProgress(){var maxTimeInSeconds=arguments.length>0&&arguments[0]!==undefined?arguments[0]:5;var _useState3=(0,_react.useState)(0),_useState4=(0,_slicedToArray2.default)(_useState3,2),elapsedTime=_useState4[0],setElapsedTime=_useState4[1];var _useState5=(0,_react.useState)(0),_useState6=(0,_slicedToArray2.default)(_useState5,2),progress=_useState6[0],setProgress=_useState6[1];(0,_react.useEffect)(function(){var intervalId=setInterval(function(){if(progress<1){setElapsedTime(function(t){return t+0.105;});}},100);return function cleanup(){clearInterval(intervalId);handleBorderActivity.handleBorderActivity();};},[]);(0,_react.useEffect)(function(){if(elapsedTime/maxTimeInSeconds<=1){setProgress(elapsedTime/maxTimeInSeconds);}else{setProgress(1);}},[elapsedTime]);return progress;};(0,_react.useEffect)(function(){var interval=setInterval(function(){var newStory=0;if(curStory!==userStories.length-1){newStory=curStory+1;setCurStory(newStory);}else{setCurStory(0);navigation.navigate('Home');}},5000);return function cleanup(){clearInterval(interval);};},[curStory]);var progress=useProgress();return _react.default.createElement(_reactNative.View,{style:styles.container,__self:this,__source:{fileName:_jsxFileName,lineNumber:70,columnNumber:9}},_react.default.createElement(_ProgressBar.default,{progress:progress,__self:this,__source:{fileName:_jsxFileName,lineNumber:71,columnNumber:13}}),_react.default.createElement(_reactNative.Image,{style:styles.image,source:{uri:userStories[curStory].imageLink},__self:this,__source:{fileName:_jsxFileName,lineNumber:72,columnNumber:13}}),_react.default.createElement(_reactNative.View,{__self:this,__source:{fileName:_jsxFileName,lineNumber:76,columnNumber:13}},_react.default.createElement(_reactNative.Text,{style:styles.Headline,__self:this,__source:{fileName:_jsxFileName,lineNumber:77,columnNumber:17}},"Headline of the story.")),_react.default.createElement(_reactNative.View,{__self:this,__source:{fileName:_jsxFileName,lineNumber:79,columnNumber:13}},_react.default.createElement(_reactNative.Text,{style:styles.opinion,__self:this,__source:{fileName:_jsxFileName,lineNumber:80,columnNumber:17}},"User opinion regarding the news.")));}var styles=_reactNative.StyleSheet.create({container:{flex:1,backgroundColor:"#264d73",alignItems:'center'},image:{marginTop:50,height:200,width:300,borderRadius:20},Headline:{marginTop:20,fontSize:15,color:"white"},opinion:{marginTop:50,fontSize:30,width:300,color:"white",textAlign:'center'}});var _default=StoryScreen;exports.default=_default;