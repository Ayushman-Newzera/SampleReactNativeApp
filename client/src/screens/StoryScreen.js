// @flow

import React, {useEffect, useState} from 'react';
import {Image, StyleSheet, Text, View} from 'react-native';
import ProgressBar from '../utilities/ProgressBar';

function StoryScreen({route, navigation}) {
  const {stories, handleBorderActivity} = route.params;
  const userStories = stories.stories;

  const [curStory, setCurStory] = useState(0);

  const useProgress = (maxTimeInSeconds = 5) => {
    const [elapsedTime, setElapsedTime] = useState(0);
    const [progress, setProgress] = useState(0);

    useEffect(() => {
      const intervalId = setInterval(() => {
        if (progress < 1) {
          setElapsedTime(t => t + 0.164);
        }
      }, 100);

      return function cleanup() {
        clearInterval(intervalId);
        handleBorderActivity.handleBorderActivity();
      };
    }, [progress]);

    useEffect(() => {
      if (elapsedTime / maxTimeInSeconds <= 1) {
        setProgress(elapsedTime / maxTimeInSeconds);
      } else {
        setProgress(1);
      }
    }, [elapsedTime, maxTimeInSeconds]);

    return progress;
  };

  useEffect(() => {
    const interval = setInterval(() => {
      let newStory = 0;
      if (curStory !== userStories.length - 1) {
        newStory = curStory + 1;
        setCurStory(newStory);
      } else {
        setCurStory(0);
        navigation.navigate('Home');
      }
    }, 5000);

    return function cleanup() {
      clearInterval(interval);
    };
  }, [curStory, navigation, userStories.length]);

  const progress = useProgress();

  return (
    <View style={styles.container}>
      <ProgressBar progress={progress} />
      <Image
        style={styles.image}
        source={{uri: userStories[curStory].imageLink}}
      />
      <View>
        <Text style={styles.Headline}>Headline of the story.</Text>
      </View>
      <View>
        <Text style={styles.opinion}>User opinion regarding the news.</Text>
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#264d73',
    alignItems: 'center',
  },
  image: {
    marginTop: 50,
    height: 200,
    width: 300,
    borderRadius: 20,
  },
  Headline: {
    marginTop: 20,
    fontSize: 15,
    color: 'white',
  },
  opinion: {
    marginTop: 50,
    fontSize: 30,
    width: 300,
    color: 'white',
    textAlign: 'center',
  },
});

export default StoryScreen;
