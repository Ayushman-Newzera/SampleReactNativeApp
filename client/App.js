/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 *
 */

import {
  ApolloClient,
  ApolloProvider,
  createHttpLink,
  InMemoryCache,
} from '@apollo/client';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import React from 'react';
import HomeScreen from './src/screens/HomeScreen';
import ProfilePictureInput from './src/screens/ProfilePictureInput';
import StoryInputScreen from './src/screens/StoryInputScreen';
import StoryScreen from './src/screens/StoryScreen';

const httpLink = createHttpLink({
  uri: 'http://192.168.1.35:4000',
});

export const client = new ApolloClient({
  link: httpLink,
  cache: new InMemoryCache(),
});

const Stack = createStackNavigator();

function App() {
  return (
    <ApolloProvider client={client}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Home" component={HomeScreen} />
          <Stack.Screen name="Stories" component={StoryScreen} />
          <Stack.Screen name="Stories Input" component={StoryInputScreen} />
          <Stack.Screen name="Profile Input" component={ProfilePictureInput} />
        </Stack.Navigator>
      </NavigationContainer>
    </ApolloProvider>
  );
}

export default App;
