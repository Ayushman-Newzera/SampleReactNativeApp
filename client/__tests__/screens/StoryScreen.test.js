import {shallow} from 'enzyme';
import React from 'react';
import StoryScreen from '../../src/screens/StoryScreen';
import {MockedProvider} from '@apollo/client/testing';

/**
 * Used to create an object of props
 * @param {object} props Custom props that needs to pass
 * @returns Props to be passed to the component/screen
 */
const createStoryScreenProps = props => ({
  navigation: {
    navigate: jest.fn(),
    pop: jest.fn(),
  },
  route: {
    params: {
      stories,
      handleBorderActivity,
    },
  },
  ...props,
});

/**
 * Constants
 */
const stories = [
  {
    imageLink:
      'https://images.unsplash.com/photo-1584714268709-c3dd9c92b378?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1099&q=80',
  },
];

const handleBorderActivity = () => {
  console.log('here');
};

describe('UselessTextInputMultiline', () => {
  it('TextInput should render correctly', () => {
    const props = createStoryScreenProps();
    const wrapper = shallow(
      <MockedProvider addTypename>
        <StoryScreen {...props} />
      </MockedProvider>,
    );

    expect(wrapper.exists()).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });
});
