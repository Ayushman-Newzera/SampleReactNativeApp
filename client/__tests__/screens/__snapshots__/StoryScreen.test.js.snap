// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UselessTextInputMultiline TextInput should render correctly 1`] = `
<MockedProvider
  addTypename={true}
>
  <ApolloProvider
    client={
      ApolloClient {
        "cache": InMemoryCache {
          "addTypename": true,
          "config": Object {
            "addTypename": true,
            "dataIdFromObject": [Function],
            "resultCaching": true,
            "typePolicies": Object {},
          },
          "data": Root {
            "canRead": [Function],
            "data": Object {},
            "getFieldValue": [Function],
            "group": CacheGroup {
              "caching": true,
              "d": [Function],
              "keyMaker": Trie {
                "makeData": [Function],
                "weakness": true,
              },
            },
            "policies": Policies {
              "cache": [Circular],
              "config": Object {
                "cache": [Circular],
                "dataIdFromObject": [Function],
                "possibleTypes": undefined,
                "typePolicies": Object {},
              },
              "fuzzySubtypes": Map {},
              "rootIdsByTypename": Object {
                "Mutation": "ROOT_MUTATION",
                "Query": "ROOT_QUERY",
                "Subscription": "ROOT_SUBSCRIPTION",
              },
              "rootTypenamesById": Object {
                "ROOT_MUTATION": "Mutation",
                "ROOT_QUERY": "Query",
                "ROOT_SUBSCRIPTION": "Subscription",
              },
              "supertypeMap": Map {},
              "toBeAdded": Object {},
              "typePolicies": Object {},
              "usingPossibleTypes": false,
            },
            "refs": Object {},
            "rootIds": Object {},
            "sharedLayerGroup": CacheGroup {
              "caching": true,
              "d": [Function],
              "keyMaker": Trie {
                "makeData": [Function],
                "weakness": true,
              },
            },
            "storageTrie": Trie {
              "makeData": [Function],
              "weakness": true,
            },
            "toReference": [Function],
          },
          "getFragmentDoc": [Function],
          "makeVar": [Function],
          "maybeBroadcastWatch": [Function],
          "optimisticData": Root {
            "canRead": [Function],
            "data": Object {},
            "getFieldValue": [Function],
            "group": CacheGroup {
              "caching": true,
              "d": [Function],
              "keyMaker": Trie {
                "makeData": [Function],
                "weakness": true,
              },
            },
            "policies": Policies {
              "cache": [Circular],
              "config": Object {
                "cache": [Circular],
                "dataIdFromObject": [Function],
                "possibleTypes": undefined,
                "typePolicies": Object {},
              },
              "fuzzySubtypes": Map {},
              "rootIdsByTypename": Object {
                "Mutation": "ROOT_MUTATION",
                "Query": "ROOT_QUERY",
                "Subscription": "ROOT_SUBSCRIPTION",
              },
              "rootTypenamesById": Object {
                "ROOT_MUTATION": "Mutation",
                "ROOT_QUERY": "Query",
                "ROOT_SUBSCRIPTION": "Subscription",
              },
              "supertypeMap": Map {},
              "toBeAdded": Object {},
              "typePolicies": Object {},
              "usingPossibleTypes": false,
            },
            "refs": Object {},
            "rootIds": Object {},
            "sharedLayerGroup": CacheGroup {
              "caching": true,
              "d": [Function],
              "keyMaker": Trie {
                "makeData": [Function],
                "weakness": true,
              },
            },
            "storageTrie": Trie {
              "makeData": [Function],
              "weakness": true,
            },
            "toReference": [Function],
          },
          "policies": Policies {
            "cache": [Circular],
            "config": Object {
              "cache": [Circular],
              "dataIdFromObject": [Function],
              "possibleTypes": undefined,
              "typePolicies": Object {},
            },
            "fuzzySubtypes": Map {},
            "rootIdsByTypename": Object {
              "Mutation": "ROOT_MUTATION",
              "Query": "ROOT_QUERY",
              "Subscription": "ROOT_SUBSCRIPTION",
            },
            "rootTypenamesById": Object {
              "ROOT_MUTATION": "Mutation",
              "ROOT_QUERY": "Query",
              "ROOT_SUBSCRIPTION": "Subscription",
            },
            "supertypeMap": Map {},
            "toBeAdded": Object {},
            "typePolicies": Object {},
            "usingPossibleTypes": false,
          },
          "storeReader": StoreReader {
            "config": Object {
              "addTypename": true,
              "cache": [Circular],
            },
            "executeSelectionSet": [Function],
            "executeSubSelectedArray": [Function],
            "knownResults": WeakMap {},
          },
          "storeWriter": StoreWriter {
            "cache": [Circular],
            "reader": StoreReader {
              "config": Object {
                "addTypename": true,
                "cache": [Circular],
              },
              "executeSelectionSet": [Function],
              "executeSubSelectedArray": [Function],
              "knownResults": WeakMap {},
            },
          },
          "txCount": 0,
          "typenameDocumentCache": Map {},
          "watchDep": [Function],
          "watches": Set {},
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": MockLink {
          "addTypename": true,
          "mockedResponsesByKey": Object {},
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "resultCaching": true,
              "typePolicies": Object {},
            },
            "data": Root {
              "canRead": [Function],
              "data": Object {},
              "getFieldValue": [Function],
              "group": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "policies": Policies {
                "cache": [Circular],
                "config": Object {
                  "cache": [Circular],
                  "dataIdFromObject": [Function],
                  "possibleTypes": undefined,
                  "typePolicies": Object {},
                },
                "fuzzySubtypes": Map {},
                "rootIdsByTypename": Object {
                  "Mutation": "ROOT_MUTATION",
                  "Query": "ROOT_QUERY",
                  "Subscription": "ROOT_SUBSCRIPTION",
                },
                "rootTypenamesById": Object {
                  "ROOT_MUTATION": "Mutation",
                  "ROOT_QUERY": "Query",
                  "ROOT_SUBSCRIPTION": "Subscription",
                },
                "supertypeMap": Map {},
                "toBeAdded": Object {},
                "typePolicies": Object {},
                "usingPossibleTypes": false,
              },
              "refs": Object {},
              "rootIds": Object {},
              "sharedLayerGroup": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "storageTrie": Trie {
                "makeData": [Function],
                "weakness": true,
              },
              "toReference": [Function],
            },
            "getFragmentDoc": [Function],
            "makeVar": [Function],
            "maybeBroadcastWatch": [Function],
            "optimisticData": Root {
              "canRead": [Function],
              "data": Object {},
              "getFieldValue": [Function],
              "group": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "policies": Policies {
                "cache": [Circular],
                "config": Object {
                  "cache": [Circular],
                  "dataIdFromObject": [Function],
                  "possibleTypes": undefined,
                  "typePolicies": Object {},
                },
                "fuzzySubtypes": Map {},
                "rootIdsByTypename": Object {
                  "Mutation": "ROOT_MUTATION",
                  "Query": "ROOT_QUERY",
                  "Subscription": "ROOT_SUBSCRIPTION",
                },
                "rootTypenamesById": Object {
                  "ROOT_MUTATION": "Mutation",
                  "ROOT_QUERY": "Query",
                  "ROOT_SUBSCRIPTION": "Subscription",
                },
                "supertypeMap": Map {},
                "toBeAdded": Object {},
                "typePolicies": Object {},
                "usingPossibleTypes": false,
              },
              "refs": Object {},
              "rootIds": Object {},
              "sharedLayerGroup": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "storageTrie": Trie {
                "makeData": [Function],
                "weakness": true,
              },
              "toReference": [Function],
            },
            "policies": Policies {
              "cache": [Circular],
              "config": Object {
                "cache": [Circular],
                "dataIdFromObject": [Function],
                "possibleTypes": undefined,
                "typePolicies": Object {},
              },
              "fuzzySubtypes": Map {},
              "rootIdsByTypename": Object {
                "Mutation": "ROOT_MUTATION",
                "Query": "ROOT_QUERY",
                "Subscription": "ROOT_SUBSCRIPTION",
              },
              "rootTypenamesById": Object {
                "ROOT_MUTATION": "Mutation",
                "ROOT_QUERY": "Query",
                "ROOT_SUBSCRIPTION": "Subscription",
              },
              "supertypeMap": Map {},
              "toBeAdded": Object {},
              "typePolicies": Object {},
              "usingPossibleTypes": false,
            },
            "storeReader": StoreReader {
              "config": Object {
                "addTypename": true,
                "cache": [Circular],
              },
              "executeSelectionSet": [Function],
              "executeSubSelectedArray": [Function],
              "knownResults": WeakMap {},
            },
            "storeWriter": StoreWriter {
              "cache": [Circular],
              "reader": StoreReader {
                "config": Object {
                  "addTypename": true,
                  "cache": [Circular],
                },
                "executeSelectionSet": [Function],
                "executeSubSelectedArray": [Function],
                "knownResults": WeakMap {},
              },
            },
            "txCount": 0,
            "typenameDocumentCache": Map {},
            "watchDep": [Function],
            "watches": Set {},
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "cache": InMemoryCache {
            "addTypename": true,
            "config": Object {
              "addTypename": true,
              "dataIdFromObject": [Function],
              "resultCaching": true,
              "typePolicies": Object {},
            },
            "data": Root {
              "canRead": [Function],
              "data": Object {},
              "getFieldValue": [Function],
              "group": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "policies": Policies {
                "cache": [Circular],
                "config": Object {
                  "cache": [Circular],
                  "dataIdFromObject": [Function],
                  "possibleTypes": undefined,
                  "typePolicies": Object {},
                },
                "fuzzySubtypes": Map {},
                "rootIdsByTypename": Object {
                  "Mutation": "ROOT_MUTATION",
                  "Query": "ROOT_QUERY",
                  "Subscription": "ROOT_SUBSCRIPTION",
                },
                "rootTypenamesById": Object {
                  "ROOT_MUTATION": "Mutation",
                  "ROOT_QUERY": "Query",
                  "ROOT_SUBSCRIPTION": "Subscription",
                },
                "supertypeMap": Map {},
                "toBeAdded": Object {},
                "typePolicies": Object {},
                "usingPossibleTypes": false,
              },
              "refs": Object {},
              "rootIds": Object {},
              "sharedLayerGroup": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "storageTrie": Trie {
                "makeData": [Function],
                "weakness": true,
              },
              "toReference": [Function],
            },
            "getFragmentDoc": [Function],
            "makeVar": [Function],
            "maybeBroadcastWatch": [Function],
            "optimisticData": Root {
              "canRead": [Function],
              "data": Object {},
              "getFieldValue": [Function],
              "group": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "policies": Policies {
                "cache": [Circular],
                "config": Object {
                  "cache": [Circular],
                  "dataIdFromObject": [Function],
                  "possibleTypes": undefined,
                  "typePolicies": Object {},
                },
                "fuzzySubtypes": Map {},
                "rootIdsByTypename": Object {
                  "Mutation": "ROOT_MUTATION",
                  "Query": "ROOT_QUERY",
                  "Subscription": "ROOT_SUBSCRIPTION",
                },
                "rootTypenamesById": Object {
                  "ROOT_MUTATION": "Mutation",
                  "ROOT_QUERY": "Query",
                  "ROOT_SUBSCRIPTION": "Subscription",
                },
                "supertypeMap": Map {},
                "toBeAdded": Object {},
                "typePolicies": Object {},
                "usingPossibleTypes": false,
              },
              "refs": Object {},
              "rootIds": Object {},
              "sharedLayerGroup": CacheGroup {
                "caching": true,
                "d": [Function],
                "keyMaker": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
              },
              "storageTrie": Trie {
                "makeData": [Function],
                "weakness": true,
              },
              "toReference": [Function],
            },
            "policies": Policies {
              "cache": [Circular],
              "config": Object {
                "cache": [Circular],
                "dataIdFromObject": [Function],
                "possibleTypes": undefined,
                "typePolicies": Object {},
              },
              "fuzzySubtypes": Map {},
              "rootIdsByTypename": Object {
                "Mutation": "ROOT_MUTATION",
                "Query": "ROOT_QUERY",
                "Subscription": "ROOT_SUBSCRIPTION",
              },
              "rootTypenamesById": Object {
                "ROOT_MUTATION": "Mutation",
                "ROOT_QUERY": "Query",
                "ROOT_SUBSCRIPTION": "Subscription",
              },
              "supertypeMap": Map {},
              "toBeAdded": Object {},
              "typePolicies": Object {},
              "usingPossibleTypes": false,
            },
            "storeReader": StoreReader {
              "config": Object {
                "addTypename": true,
                "cache": [Circular],
              },
              "executeSelectionSet": [Function],
              "executeSubSelectedArray": [Function],
              "knownResults": WeakMap {},
            },
            "storeWriter": StoreWriter {
              "cache": [Circular],
              "reader": StoreReader {
                "config": Object {
                  "addTypename": true,
                  "cache": [Circular],
                },
                "executeSelectionSet": [Function],
                "executeSubSelectedArray": [Function],
                "knownResults": WeakMap {},
              },
            },
            "txCount": 0,
            "typenameDocumentCache": Map {},
            "watchDep": [Function],
            "watches": Set {},
          },
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "fetchCancelFns": Map {},
          "inFlightLinkObservables": Map {},
          "link": MockLink {
            "addTypename": true,
            "mockedResponsesByKey": Object {},
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": true,
              "config": Object {
                "addTypename": true,
                "dataIdFromObject": [Function],
                "resultCaching": true,
                "typePolicies": Object {},
              },
              "data": Root {
                "canRead": [Function],
                "data": Object {},
                "getFieldValue": [Function],
                "group": CacheGroup {
                  "caching": true,
                  "d": [Function],
                  "keyMaker": Trie {
                    "makeData": [Function],
                    "weakness": true,
                  },
                },
                "policies": Policies {
                  "cache": [Circular],
                  "config": Object {
                    "cache": [Circular],
                    "dataIdFromObject": [Function],
                    "possibleTypes": undefined,
                    "typePolicies": Object {},
                  },
                  "fuzzySubtypes": Map {},
                  "rootIdsByTypename": Object {
                    "Mutation": "ROOT_MUTATION",
                    "Query": "ROOT_QUERY",
                    "Subscription": "ROOT_SUBSCRIPTION",
                  },
                  "rootTypenamesById": Object {
                    "ROOT_MUTATION": "Mutation",
                    "ROOT_QUERY": "Query",
                    "ROOT_SUBSCRIPTION": "Subscription",
                  },
                  "supertypeMap": Map {},
                  "toBeAdded": Object {},
                  "typePolicies": Object {},
                  "usingPossibleTypes": false,
                },
                "refs": Object {},
                "rootIds": Object {},
                "sharedLayerGroup": CacheGroup {
                  "caching": true,
                  "d": [Function],
                  "keyMaker": Trie {
                    "makeData": [Function],
                    "weakness": true,
                  },
                },
                "storageTrie": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
                "toReference": [Function],
              },
              "getFragmentDoc": [Function],
              "makeVar": [Function],
              "maybeBroadcastWatch": [Function],
              "optimisticData": Root {
                "canRead": [Function],
                "data": Object {},
                "getFieldValue": [Function],
                "group": CacheGroup {
                  "caching": true,
                  "d": [Function],
                  "keyMaker": Trie {
                    "makeData": [Function],
                    "weakness": true,
                  },
                },
                "policies": Policies {
                  "cache": [Circular],
                  "config": Object {
                    "cache": [Circular],
                    "dataIdFromObject": [Function],
                    "possibleTypes": undefined,
                    "typePolicies": Object {},
                  },
                  "fuzzySubtypes": Map {},
                  "rootIdsByTypename": Object {
                    "Mutation": "ROOT_MUTATION",
                    "Query": "ROOT_QUERY",
                    "Subscription": "ROOT_SUBSCRIPTION",
                  },
                  "rootTypenamesById": Object {
                    "ROOT_MUTATION": "Mutation",
                    "ROOT_QUERY": "Query",
                    "ROOT_SUBSCRIPTION": "Subscription",
                  },
                  "supertypeMap": Map {},
                  "toBeAdded": Object {},
                  "typePolicies": Object {},
                  "usingPossibleTypes": false,
                },
                "refs": Object {},
                "rootIds": Object {},
                "sharedLayerGroup": CacheGroup {
                  "caching": true,
                  "d": [Function],
                  "keyMaker": Trie {
                    "makeData": [Function],
                    "weakness": true,
                  },
                },
                "storageTrie": Trie {
                  "makeData": [Function],
                  "weakness": true,
                },
                "toReference": [Function],
              },
              "policies": Policies {
                "cache": [Circular],
                "config": Object {
                  "cache": [Circular],
                  "dataIdFromObject": [Function],
                  "possibleTypes": undefined,
                  "typePolicies": Object {},
                },
                "fuzzySubtypes": Map {},
                "rootIdsByTypename": Object {
                  "Mutation": "ROOT_MUTATION",
                  "Query": "ROOT_QUERY",
                  "Subscription": "ROOT_SUBSCRIPTION",
                },
                "rootTypenamesById": Object {
                  "ROOT_MUTATION": "Mutation",
                  "ROOT_QUERY": "Query",
                  "ROOT_SUBSCRIPTION": "Subscription",
                },
                "supertypeMap": Map {},
                "toBeAdded": Object {},
                "typePolicies": Object {},
                "usingPossibleTypes": false,
              },
              "storeReader": StoreReader {
                "config": Object {
                  "addTypename": true,
                  "cache": [Circular],
                },
                "executeSelectionSet": [Function],
                "executeSubSelectedArray": [Function],
                "knownResults": WeakMap {},
              },
              "storeWriter": StoreWriter {
                "cache": [Circular],
                "reader": StoreReader {
                  "config": Object {
                    "addTypename": true,
                    "cache": [Circular],
                  },
                  "executeSelectionSet": [Function],
                  "executeSubSelectedArray": [Function],
                  "knownResults": WeakMap {},
                },
              },
              "txCount": 0,
              "typenameDocumentCache": Map {},
              "watchDep": [Function],
              "watches": Set {},
            },
            "client": [Circular],
          },
          "mutationIdCounter": 1,
          "mutationStore": Object {},
          "onBroadcast": [Function],
          "queries": Map {},
          "queryDeduplication": true,
          "queryIdCounter": 1,
          "requestIdCounter": 1,
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "typeDefs": undefined,
        "version": "3.3.20",
        "watchQuery": [Function],
      }
    }
  >
    <StoryScreen
      navigation={
        Object {
          "navigate": [MockFunction],
          "pop": [MockFunction],
        }
      }
      route={
        Object {
          "params": Object {
            "handleBorderActivity": Object {
              "handleBorderActivity": [Function],
            },
            "stories": Object {
              "stories": Array [
                Object {
                  "imageLink": "https://www.google.com/url?sa=i&url=https%3A%2F%2Funsplash.com%2Fs%2Fphotos%2Fcars&psig=AOvVaw2tJrsSDJK_jfSHbgnx7FiX&ust=1625234888760000&source=images&cd=vfe&ved=0CAcQjRxqFwoTCPDqrPOFwvECFQAAAAAdAAAAABAM",
                },
              ],
            },
          },
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#264d73",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#264d73",
              "flex": 1,
            }
          }
        >
          <ProgressBar
            progress={0}
          >
            <View
              style={
                Object {
                  "backgroundColor": "rgba(0, 0, 0, 0)",
                  "borderColor": "#000",
                  "borderRadius": 5,
                  "borderWidth": 2,
                  "height": 11,
                  "marginTop": 50,
                  "width": "80%",
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "rgba(0, 0, 0, 0)",
                    "borderColor": "#000",
                    "borderRadius": 5,
                    "borderWidth": 2,
                    "height": 11,
                    "marginTop": 50,
                    "width": "80%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "white",
                        "borderRadius": 4,
                        "borderWidth": 1.5,
                        "height": 9,
                        "marginTop": -1,
                      },
                      Object {
                        "width": "0%",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "backgroundColor": "white",
                          "borderRadius": 4,
                          "borderWidth": 1.5,
                          "height": 9,
                          "marginTop": -1,
                        },
                        Object {
                          "width": "0%",
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </ProgressBar>
          <Image
            source={
              Object {
                "uri": "https://www.google.com/url?sa=i&url=https%3A%2F%2Funsplash.com%2Fs%2Fphotos%2Fcars&psig=AOvVaw2tJrsSDJK_jfSHbgnx7FiX&ust=1625234888760000&source=images&cd=vfe&ved=0CAcQjRxqFwoTCPDqrPOFwvECFQAAAAAdAAAAABAM",
              }
            }
            style={
              Object {
                "borderRadius": 20,
                "height": 200,
                "marginTop": 50,
                "width": 300,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://www.google.com/url?sa=i&url=https%3A%2F%2Funsplash.com%2Fs%2Fphotos%2Fcars&psig=AOvVaw2tJrsSDJK_jfSHbgnx7FiX&ust=1625234888760000&source=images&cd=vfe&ved=0CAcQjRxqFwoTCPDqrPOFwvECFQAAAAAdAAAAABAM",
                }
              }
              style={
                Object {
                  "borderRadius": 20,
                  "height": 200,
                  "marginTop": 50,
                  "width": 300,
                }
              }
            />
          </Image>
          <View>
            <View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 15,
                    "marginTop": 20,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 15,
                      "marginTop": 20,
                    }
                  }
                >
                  Headline of the story.
                </Text>
              </Text>
            </View>
          </View>
          <View>
            <View>
              <Text
                style={
                  Object {
                    "color": "white",
                    "fontSize": 30,
                    "marginTop": 50,
                    "textAlign": "center",
                    "width": 300,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "color": "white",
                      "fontSize": 30,
                      "marginTop": 50,
                      "textAlign": "center",
                      "width": 300,
                    }
                  }
                >
                  User opinion regarding the news.
                </Text>
              </Text>
            </View>
          </View>
        </View>
      </View>
    </StoryScreen>
  </ApolloProvider>
</MockedProvider>
`;
